src/app/modules/startup/controller.nim

```diff
@@ -433,56 +420,48 @@ proc importAccountAndLogin*(self: Controller, storeToKeychain: bool, recoverAcco
     self.tmpProfileImageDetails.url,
     self.tmpProfileImageDetails.cropRectangle,
   )
+
   self.processCreateAccountResult(error, storeToKeychain)
 
-proc storeKeycardAccountAndLogin*(self: Controller, storeToKeychain: bool, newKeycard: bool) =
-  if self.importMnemonic():
-    self.delegate.moveToLoadingAppState()
-    if newKeycard:
-      self.delegate.storeDefaultKeyPairForNewKeycardUser()
-      self.storeMetadataForNewKeycardUser()
-    else:
-      self.syncKeycardBasedOnAppWalletStateAfterLogin()
-    let (_, flowEvent) = self.keycardService.getLastReceivedKeycardData() # we need this to get the correct instanceUID
-    self.accountsService.setupAccountKeycard(flowEvent, self.tmpDisplayName, useImportedAcc = true)
-    self.setupKeychain(storeToKeychain)
-  else:
-    error "an error ocurred while importing mnemonic"
+# NOTE: Called during FirstRunNewUserNewKeycardKeys and FirstRunNewUserImportSeedPhraseIntoKeycard
+# WARNING: Reuse `importAccountAndLogin` with custom parameters
+proc storeKeycardAccountAndLogin*(self: Controller, storeToKeychain: bool, newKeycard: bool = true) =
+  self.delegate.moveToLoadingAppState()
+  self.storeMetadataForNewKeycardUser()
+  let (_, flowEvent) = self.keycardService.getLastReceivedKeycardData()
+  let error = self.accountsService.importAccountAndLogin(
+    self.tmpSeedPhrase, 
+    "", # Keep an empty password. For keycard it will be substituted with`encryption.publicKey` in status-go
+    false, 
```

Comment:
Can you put the param name here just because it's cool

Reply:
ca879b7facedaf5c54372ee059b3d4fe913d715a